src/lib/swapper/swappers/OneInchSwapper/executeTrade/executeTrade.ts

```diff
@@ -57,13 +58,18 @@ export async function executeTrade(
       })
     }
 
+    const { average } = await adapter.getGasFeeData()
+
+    // TODO: networkFeeCryptoBaseUnit should be a gas limit as well so this isnt necessary
```

Comment:
If I get this correctly, it is currently a gasLimit? If so we should definitely rename it, as the naming implies that this is already a fee in wei which indeed would make the calculation wrong

Reply:
Apologies, my comment wasn't clear. 

The `networkFeeCryptoBaseUnit` is already a fee in wei but executing the trade requires a gas limit (price can be pulled from react). The get the limit we need to divide the fee by the price which may have changed - invalidating the limit provided in the quote.

It seems to be a similar issue with the approvals where we were conflating the idea of display concerns and transaction execution requirements. `networkFeeCryptoBaseUnit` is really intended for display purposes and can be computed in the UI layer using a gas limit on the quote.

In the case of approvals, we can compute the gas limit and price at the time of the approval, which makes storing this information on the quote redundant, but for the actual trade we should store the gas limit on the quote and only compute the fee at the time of displaying it or executing the trade using the current market price.

Reply:
Oh you're absolutely right. We definitely shouldn't store a view-layer concern in a quote (though nothing necessarily wrong to store it derived data to ease the view-layer boilerplate, but I would definitely be inclined towards only storing atomic fields, not derived ones).

`gasLimit` is all we should eventually store 🙏🏽